### PR TITLE
[5.2] Allow Arr::get to work with ArrayAccess objects.

### PR DESCRIPTION
**Pull request by donnysim**
Allows Arr::get to work with ArrayAccess objects.

### CODE REVIEW

File: src/Illuminate/Support/Arr.php [hunk 1]
```diff
@@ -233,7 +233,8 @@ public static function get($array, $key, $default = null)
         }
 
         foreach (explode('.', $key) as $segment) {
-            if (! is_array($array) || ! array_key_exists($segment, $array)) {
+            if ((! is_array($array) || ! array_key_exists($segment, $array)) &&
+                (! $array instanceof \ArrayAccess || ! $array->offsetExists($segment))) {
```
Comment (lucasmichot) [inline src/Illuminate/Support/Arr.php]: You should import `\ArrayAccess`

File: src/Illuminate/Support/Arr.php [hunk 2]
```diff
@@ -233,7 +234,8 @@ public static function get($array, $key, $default = null)
         }
 
         foreach (explode('.', $key) as $segment) {
-            if (! is_array($array) || ! array_key_exists($segment, $array)) {
+            if ((! is_array($array) || ! array_key_exists($segment, $array)) &&
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Arr.php]: Better to just call isset.

Comment (donnysim) [inline src/Illuminate/Support/Arr.php]: Wouldn't that introduce a false positive then? If I make my default value let's say `false` and the the array key exists, but is `null`, with `isset` I won't get `null`, but `false` as a result instead.

**Comment by vlakoff** [comment]
I think it should be implemented in `Arr::has` as well, for consistency.

**Pull request merged**